helm/chart-operator/values.yaml

```diff
@@ -0,0 +1,35 @@
+image:
+  registry: "quay.io"
+  repository: "giantswarm/chart-operator"
+  tag: "[[ .Version ]]"
+pod:
+  user:
+    id: 1000
+  group:
+    id: 1000
+  port: 8000
+  replicas: 1
+project:
+  name: "chart-operator"
+# Resource names are truncated to 47 characters. Kubernetes allows 63 characters
+# limit for resource names. When pods for deployments are created they have
+# additional 16 characters suffix, e.g. "-957c9d6ff-pkzgw" and we want to have
+# room for those suffixes.
+#
+# NOTE: All values under resource key need to be used with `tpl` to render them
+# correctly in the templates. This is because helm doesn't template values.yaml
+# file and it has to be a valid json. Example usage:
+#
+#     {{ tpl .Values.resource.default.name . }}.
+#
+resource:
+  clusterDNSIP: 172.31.0.10
```

Comment:
I think this should go out of `resource`. I'd like to keep here only resource names. This doesn't also have to be wrapped with `tpl`.

Reply:
Agree with @kopiczko but also we need to be very careful here because `clusterDNSIP` is set by cluster-operator when it installs chart-operator.

IIRC this is so chart-operator can install CoreDNS in new clusters. If we change the values structure this will probably fail.

https://github.com/giantswarm/cluster-operator/blob/9e12a2aedf6647c4aeaa6ef84b4fb0326205e199/pkg/v18/resource/chartoperator/desired.go#L25

Reply:
@rossf7 But that resource using `chart-operator-chart` helm chart so we are okay to move it.

Reply:
> I think this should go out of resource. I'd like to keep here only resource names. 

Like Pawel says this block is for the resource naming and `clusterDNSIP` isn't part of that. So I'd like us to keep the original structure and make this a top level value.

Reply:
In the chartoperator resource in app-operator we will need to set this value. Like we do now in cluster-operator.

https://github.com/giantswarm/cluster-operator/blob/9e12a2aedf6647c4aeaa6ef84b4fb0326205e199/pkg/v18/resource/chartoperator/desired.go#L25